geomesa-kafka/geomesa-kafka-confluent/src/main/scala/org/locationtech/geomesa/kafka/confluent/ConfluentKafkaDataStoreFactory.scala

```diff
@@ -54,14 +62,50 @@ object ConfluentKafkaDataStoreFactory extends GeoMesaDataStoreInfo with LazyLogg
   override val DisplayName = "Confluent Kafka (GeoMesa)"
   override val Description = "Confluent Apache Kafka\u2122 distributed log"
 
-  val SchemaRegistryUrl = new GeoMesaParam[URL]("kafka.schema.registry.url", "URL to a confluent schema registry server, used to read Confluent schemas (experimental)")
+  val SchemaRegistryUrl =
+    new GeoMesaParam[URL](
+      "kafka.schema.registry.url",
+      "URL to a confluent schema registry server, used to read Confluent schemas (experimental)",
+      optional = false
+    )
+
+  val SchemaOverrides =
+    new GeoMesaParam[String](
+      "kafka.schema.overrides",
+      "Typesafe configuration defining a map from topic name to schema override (experimental)",
+      optional = true,
+      largeText = true
+    )
 
   override val ParameterInfo: Array[GeoMesaParam[_ <: AnyRef]] =
-    KafkaDataStoreFactory.ParameterInfo.+:(SchemaRegistryUrl)
+    KafkaDataStoreFactory.ParameterInfo ++ Array(SchemaRegistryUrl, SchemaOverrides)
 
   override def canProcess(params: java.util.Map[String, _ <: java.io.Serializable]): Boolean = {
-    KafkaDataStoreFactoryParams.Brokers.exists(params) &&
-        KafkaDataStoreFactoryParams.Zookeepers.exists(params) &&
-        SchemaRegistryUrl.exists(params)
+    KafkaDataStoreParams.Brokers.exists(params) &&
+      KafkaDataStoreParams.Zookeepers.exists(params) &&
+      SchemaRegistryUrl.exists(params)
+  }
+
+  private[confluent] def parseSchemaOverrides(schemaOverridesConfig: String): Map[String, (SimpleFeatureType, Schema)] = {
+    try {
+      Option(schemaOverridesConfig)
+        .map(ConfigFactory.parseString(_).resolve())
+        .filter(_.hasPath("schemas"))
```

Comment:
should this be an error if there's no 'schemas' defined?

Reply:
I thought it would be fine for the map to just be empty, I have a unit test for that case.

Reply:
i was thinking if you had a typo in the config or something, it would just fail silently and might be hard to find out why. maybe we could just log a warning if there is a config provided but there's no 'schemas' key present, although an exception would be more obvious ;)

Reply:
Gotcha, I'll add that